### PR TITLE
support deleted files in kusion deps cmd

### PR DESCRIPTION
**Pull request by amyXia1994**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [X] N
- [ ] Y 

#### 2. What is the scope of this PR (e.g. component or file name):

- pkg/kusionctl/cmd/deps/options.go
- go.mod: In order to update the latest kclvm-go deps API, the kclvm-go version in go.mod is updated to v0.4.3-alpha.1

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

before the PR, the in-existing file paths as ```--focus``` option is not allowed in using the ```kusion deps``` command to find downstream stacks/projects and the in-existing file paths will cause an "invalid focus path" error.
This PR support deleted files in ```kusion deps --focus``` option, and slightly improves the performance of that command.

- [X] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [X] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [X] Unit test
- [ ] Integration test
- [X] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

This PR adds some unit tests and benchmarks on the core logic about down stream stacks/projects analyzing.

This PR brings some reduction on memory usage result based on the testdata and brings about 50% time and memory consumption base on a 2000+ stacks Konfig repository.

benchstat result on processing downstream stacks/projects of the testdata:
```
DownStream-4    5.73ms ±40%    4.52ms ±14%     ~     (p=0.222 n=5+5)

name          old alloc/op   new alloc/op   delta
DownStream-4     241kB ± 0%     196kB ± 0%  -18.81%  (p=0.008 n=5+5)

name          old allocs/op  new allocs/op  delta
DownStream-4     2.04k ± 0%     1.58k ± 0%  -22.44%  (p=0.008 n=5+5)
```

benchstat result on procession a 2000+ stacks Konfig repository:
```
name                             old time/op    new time/op    delta
KonfigTotal_siteBuild_simple-4      15.7s ±16%      6.9s ±36%  -56.00%  (p=0.008 n=5+5)                                                                                        
KonfigTotal_siteBuild_complex-4     18.0s ±14%      9.6s ± 9%  -46.79%  (p=0.016 n=5+4)

name                             old alloc/op   new alloc/op   delta
KonfigTotal_siteBuild_simple-4     1.04GB ± 0%    0.33GB ± 0%  -68.13%  (p=0.008 n=5+5)
KonfigTotal_siteBuild_complex-4    1.04GB ± 0%    0.42GB ± 0%  -59.65%  (p=0.008 n=5+5)

name                             old allocs/op  new allocs/op  delta
KonfigTotal_siteBuild_simple-4      7.12M ± 0%     2.78M ± 0%  -60.88%  (p=0.008 n=5+5)
KonfigTotal_siteBuild_complex-4     7.12M ± 0%     3.27M ± 0%  -53.99%  (p=0.008 n=5+5)
```

#### 6. Release note

```release-note
support deleted files in ```--focus```option. of the ```kusion deps``` command.

```

### CODE REVIEW

File: pkg/kusionctl/cmd/deps/options.go [hunk 1]
```diff
@@ -63,120 +93,176 @@ func (o *DepsOptions) Validate() error {
...
-func (o *DepsOptions) Run() error {
+func (o *DepsOptions) Run() (err error) {
 	workDir, err := filepath.Abs(o.workDir)
 	if err != nil {
-		return err
+		return
 	}
 	o.workDir = workDir
 	switch o.Direct {
+	// list upstream files of the focus files
```
Comment (elliotxx) [inline pkg/kusionctl/cmd/deps/options.go:103]: 注释首字母需要大写

Comment (amyXia1994) [reply pkg/kusionctl/cmd/deps/options.go:103]: It's a one-line comment (not a doc comment), I think using lower case is fine :) for example, see built-in lib crypto source code: https://github.com/golang/go/blob/master/src/crypto/sha1/sha1.go#L219

Comment (amyXia1994) [reply pkg/kusionctl/cmd/deps/options.go]: All comments updated accordingly

**Review by elliotxx — APPROVED** [review]
lgtm

**Review by howieyuen — APPROVED** [review]

**Pull request merged**